test/js/node/path/parse-format.test.js

```diff
@@ -0,0 +1,223 @@
+            if (!expectedKeys.includes(key) || actual[key] !== expected[key]) {
+              failed = true;
+              break;
+            }
+          }
+        }
+        if (failed) failures.push(`\n${message}`);
+      });
+    });
+    assert.strictEqual(failures.length, 0, failures.join(""));
+
+    function checkErrors(path) {
+      errors.forEach(({ method, input }) => {
+        assert.throws(
+          () => {
+            path[method].apply(path, input);
+          },
+          {
+            code: "ERR_INVALID_ARG_TYPE",
+            name: "TypeError",
+          },
+        );
+      });
+    }
+
+    function checkParseFormat(path, paths) {
+      paths.forEach(([element, root]) => {
+        const output = path.parse(element);
+        assert.strictEqual(typeof output.root, "string");
+        assert.strictEqual(typeof output.dir, "string");
+        assert.strictEqual(typeof output.base, "string");
+        assert.strictEqual(typeof output.ext, "string");
+        assert.strictEqual(typeof output.name, "string");
+        assert.strictEqual(path.format(output), element);
+        assert.strictEqual(output.root, root);
+        assert(output.dir.startsWith(output.root));
+        assert.strictEqual(output.dir, output.dir ? path.dirname(element) : "");
+        assert.strictEqual(output.base, path.basename(element));
+        assert.strictEqual(output.ext, path.extname(element));
+      });
+    }
+
+    function checkSpecialCaseParseFormat(path, testCases) {
+      testCases.forEach(([element, expect]) => {
+        assert.deepStrictEqual(path.parse(element), expect);
+      });
+    }
+
+    function checkFormat(path, testCases) {
+      testCases.forEach(([element, expect]) => {
+        assert.strictEqual(path.format(element), expect);
+      });
+
+      [null, undefined, 1, true, false, "string"].forEach(pathObject => {
+        assert.throws(
+          () => {
+            path.format(pathObject);
+          },
+          {
+            code: "ERR_INVALID_ARG_TYPE",
+            name: "TypeError",
+            // TODO: Make our error messages use util.inspect like Node:
+            // https://github.com/nodejs/node/blob/68885d512640556ba95b18f5ab2e0b9e76013399/lib/internal/errors.js#L1370-L1440
+            // https://github.com/nodejs/node/blob/68885d512640556ba95b18f5ab2e0b9e76013399/test/common/index.js#L815
+            //
+            // Node's error message template is:
+            //  `The "pathObject" argument must be of type object. Received ${inspect(input, { depth: -1 })}`
+            //
+            // For example, when we throw for path.format(null) our error message is:
+            //   The "pathObject" property must be of type object, got object
+            //
+            // While Node's error message is:
+            //   The "pathObject" argument must be of type object. Received nullå
+            message: `"pathObject" property must be of type object, got ${typeof pathObject}`,
```

Comment:
👆 TODO for error message compliance. Node provides a bit more-helpful messages around data types.